Dockerfile

```diff
@@ -136,6 +139,22 @@ COPY --chown=$MAMBA_USER:$MAMBA_USER micro_environment.yml ${TETHYS_HOME}/tethys
 
 WORKDIR ${TETHYS_HOME}/tethys
 
+# Set the versions of Django, Channels, and Daphne if provided in environment.tyml and micro_environment.yml
+RUN if [ -n "$DJANGO_VERSION" ]; then \
+  sed -i "s/- django>=3.2,<6/- django==${DJANGO_VERSION}/" environment.yml micro_environment.yml; \
+  fi && \
+  if [ -n "$DJANGO_CHANNELS_VERSION" ]; then \
+  sed -i "s/- channels/- channels==${DJANGO_CHANNELS_VERSION}/" environment.yml micro_environment.yml; \
+  else \
+  sed -i "s/- channels/- channels/" environment.yml micro_environment.yml; \
+  fi && \
+  if [ -n "$DAPHNE_VERSION" ]; then \
+  sed -i "s/- daphne/- daphne==${DAPHNE_VERSION}/" environment.yml micro_environment.yml; \
+  else \
+  sed -i "s/- daphne/- daphne/" environment.yml micro_environment.yml; \
```

Comment:
The else probably isn't necessary here either.

Reply:
No, it is not necessary, and it was removed